python/ray/node.py

```diff
@@ -249,12 +250,25 @@ def _init_temp(self, redis_client):
 
     def get_resource_spec(self):
         """Resolve and return the current resource spec for the node."""
+        resources = {}
+        try:
+            env_string = os.getenv("RAY_RESOURCES")
+            if env_string:
+                resources = json.loads(env_string)
+        except Exception:
```

Comment:
Catch json exception instead?

Reply:
Can you actually remove this try block? It's preferable to fail startup if the config is mis-formatted.